### PR TITLE
feat: implement object stores

### PR DESCRIPTION
**Pull request by yihuang**
## Description

Module parameters are loaded multiple times for each tx, although the raw bytes are cached, the decoding process is not.

Support an object store that is transient stores for `interface{}`, applications can cache objects directly, without doing unnecessary encoding/decoding.

Caching module parameters is an important use case for this, there are also many use cases out there.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ## Bug Fixes
 
 * (crypto) [#19691](https://github.com/cosmos/cosmos-sdk/pull/19745) Fix tx sign doesn't throw an error when incorrect Ledger is used.
+* (store) [#]() Support object store which is a transient store support `interface{}`.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (store) [#206](https://github.com/crypto-org-chain/cosmos-sdk/pull/206) Support object store which is a transient store support `interface{}`.
```

File: store/types/store.go [hunk 2]
```diff
@@ -332,6 +365,7 @@ const (
 	StoreTypeMemory
 	StoreTypeSMT
 	StoreTypePersistent
+	StoreTypeObject
 )
 
 func (st StoreType) String() string {
```
Comment (mmsqe) [inline store/types/store.go:371]: miss `StoreTypeObject` here

**Review by mmsqe — COMMENTED** [review]
need use `storetypes.GKVStore[V]` in [runtime](https://github.com/crypto-org-chain/cosmos-sdk/blob/6fbc72701fc7ae12970b5ce477e018fa310a26a2/runtime/store.go#L100) right?

File: store/types/store.go (hunk 2 above)
Comment (mmsqe) [inline store/types/store.go:368]: miss `StoreTypeObject` here

**Comment by yihuang** [comment]
> need use `storetypes.GKVStore[V]` in [runtime](https://github.com/crypto-org-chain/cosmos-sdk/blob/6fbc72701fc7ae12970b5ce477e018fa310a26a2/runtime/store.go#L100) right?

dependers to be changed in other PR.

File: store/types/store.go (hunk 2 above)
Comment (yihuang) [reply store/types/store.go:368]: fixed

**Review by mmsqe — APPROVED** [review]

File: baseapp/baseapp.go [hunk 3]
```diff
@@ -344,6 +344,17 @@ func (app *BaseApp) MountMemoryStores(keys map[string]*storetypes.MemoryStoreKey
 	}
 }
 
+// MountObjectStores mounts all in-memory KVStores with the BaseApp's internal
```
Comment (yihuang) [inline baseapp/baseapp.go]:
```suggestion
// MountObjectStores mounts all transient object stores with the BaseApp's internal
```

**Pull request merged**